pkg/kubectl/proxy_server.go

```diff
@@ -34,7 +34,7 @@ import (
 
 const (
 	DefaultHostAcceptRE   = "^localhost$,^127\\.0\\.0\\.1$,^\\[::1\\]$"
-	DefaultPathAcceptRE   = "^/.*"
+	DefaultPathAcceptRE   = "^.*"
```

Comment:
It won't, since the path and host are matched individually (see https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/proxy_server.go#L105)